src/Bicep.Core/TypeSystem/NamedObjectType.cs

```diff
@@ -9,11 +9,16 @@ namespace Bicep.Core.TypeSystem
     public class NamedObjectType : ObjectType
     {
         public NamedObjectType(string name, TypeSymbolValidationFlags validationFlags, IEnumerable<TypeProperty> properties, ITypeReference? additionalPropertiesType, TypePropertyFlags additionalPropertiesFlags = TypePropertyFlags.None, IEnumerable<FunctionOverload>? functions = null)
+            : this(name, validationFlags, properties, additionalPropertiesType, additionalPropertiesFlags, FunctionResolver.Create(functions)) {
+            
+        }
+
+        public NamedObjectType(string name, TypeSymbolValidationFlags validationFlags, IEnumerable<TypeProperty> properties, ITypeReference? additionalPropertiesType, TypePropertyFlags additionalPropertiesFlags, FunctionResolver functionResolver)
```

Comment:
During building process, there are 2 places where we convert one NamedObjectType instance into a fresh one and the MethodResolver was being lost. To properly surface methods added on a type in AzResourceTypeProvider I needed to preserve it.